### PR TITLE
Fix broken links [#1569]

### LINKED ISSUE #1569: Fix broken links in docs
In order to [implement automated link checking in RTD as part of CI](https://github.com/nextstrain/.github/pull/104), it is necessary to manually audit broken links in this repo first.

### PR DESCRIPTION
**Pull request by genehack**
## Description of proposed changes

* adds linkchecker config to `augur`
* rewrite some markdown doc files into RST to facilitate linking in the "proper" way
* fix broken links as reported by linkchecker

## Related issue(s)

Closes #1569 

## Checklist

- [x] Automated checks pass
- [x] [Check][1] if you need to add a changelog message
- [x] [Check][2] if you need to add tests
- [x] [Check][3] if you need to update docs

[1]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#updating-the-changelog
[2]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#testing
[3]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#when-to-update

### CODE REVIEW

File: docs/releases/migrating-v5-v6.rst
Comment (victorlin) [inline docs/releases/migrating-v5-v6.rst:1]: Could you provide details on what/how you did the conversion in 003dce75520d5ca190586c6da27e072fb3e6ae01 (and other conversion commits)? Comparing [before](https://docs.nextstrain.org/projects/augur/en/latest/releases/migrating-v5-v6.html) and [after](https://nextstrain--1570.org.readthedocs.build/projects/augur/en/1570/releases/migrating-v5-v6.html), I see you've added some improvements such as `.. note::` which are great, but some other parts have not been converted properly:

https://github.com/nextstrain/augur/blob/a6b14d7b9a00e9bcd259dfca9676d20ed3caca4b/docs/releases/migrating-v5-v6.rst#L729-L733

https://github.com/nextstrain/augur/blob/a6b14d7b9a00e9bcd259dfca9676d20ed3caca4b/docs/releases/migrating-v5-v6.rst#L618

**Review by victorlin — CHANGES_REQUESTED** [review]

File: docs/releases/migrating-v5-v6.rst
Comment (genehack) [reply docs/releases/migrating-v5-v6.rst:1]: The conversion was manual.

Comment (genehack) [reply docs/releases/migrating-v5-v6.rst:1]: If there are specific changes, please suggest them.

File: docs/releases/migrating-v5-v6.rst [hunk 1]
```diff
@@ -0,0 +1,738 @@
...
+There are a couple of exceptions to this:
+
+* There is no way to set default display views using command line
+  only, so using "display_defaults" in your config file will set this.
+* There is no way to modify the default filters displayed when using
+  command line only, so using "filters" in your config file will set
+  this.
+* If you set color-by options in command-line using
+  `--metadata-color-by` _and_ pass in a config file, only the things
+  listed in `--metadata-color-by` will be coloring options, but if
```
Comment (victorlin) [inline docs/releases/migrating-v5-v6.rst:403]: Fix syntax

```suggestion
* If you set color-by options in command-line using
  ``--metadata-color-by`` *and* pass in a config file, only the things
  listed in ``--metadata-color-by`` will be coloring options, but if
```

File: docs/releases/migrating-v5-v6.rst
Comment (victorlin) [reply docs/releases/migrating-v5-v6.rst:1]: I looked at the before and after renderings up through the "Config file priority" section and made specific suggestions: https://github.com/nextstrain/augur/pull/1570#pullrequestreview-2236543656

The rest of the file needs some work too but I haven't gone through it.

I didn't realize that you did the conversion manually. That seems like a lot of work and while looking at the diffs I noticed a lot of nice improvements, so thank you for making those. Consider this non-blocking, but changes would be more digest-able if the commit was broken up into two parts: 1 direct conversion and 1 improvements. This is what I've done in the past (see https://github.com/nextstrain/augur/compare/b20df48b27e989c82fad8619f06f7cde9935083c...79ac0c848d0700d6f28f3c10b00d8830878f2fbd)

File: docs/releases/migrating-v5-v6.rst [hunk 2]
```diff
@@ -0,0 +1,738 @@
...
+entropy``).
+
+.. note:: If you want to display the frequencies panel, you must
+          specify "frequencies" *and* ensure a tip frequency file is
+          available for ``auspice`` to access.
+
+Traits
+******
+
+*(:ref:`What's a trait? traits`)* Traits will become coloring options
```
[the remark this answers is not in the document]
Comment (genehack) [reply docs/releases/migrating-v5-v6.rst:299]: Fixed. 

Note that it was necessary to remove the italics from this link, as [docutils doesn't support nested markup](https://docutils.sourceforge.io/FAQ.html#is-nested-inline-markup-possible).

File: docs/releases/migrating-v5-v6.rst [hunk 3]
```diff
@@ -0,0 +1,738 @@
...
+.. note:: You can't specify the title or type of a colouring option
+          using just command-line - but `export v2` will make its best
+          guess using the following rules:_ _Excluding missing data,
+          if a trait contains only 'True', 'False', 'Yes', 'No', '0'
+          or '1', it will be set to 'boolean.'_ _If it contains only
+          numbers (integers and/or decimals), it will be set to
+          'continuous.'_ _Otherwise, it will be set as 'discrete.'_
+          _If you want to have more control over how your trait is
+          interpreted, you should use a config file (see
+          :ref:`below <using-a-config-file-to-customise-the-visualisation>`.
```
[the remark this answers is not in the document]
Comment (genehack) [reply docs/releases/migrating-v5-v6.rst:334]: I adjusted the formatting. I think the "note" callout box is sufficient highlighting and the italics in the previous version can be dispensed with.

**Comment by genehack** [comment]
@victorlin thanks for the detailed review. I've fixed everything you noted, and done a further side-by-side comparison of both the v5->v6 and the v6 docs — I think these are now as close as possible to the originals, and generally better in a couple respects (e.g., the side by side config in the new version is IMO much nicer).

File: docs/releases/migrating-v5-v6.rst [hunk 4]
```diff
@@ -0,0 +1,750 @@
...
+* A required "key", which is used to lookup the values via node-data
+  JSONs or other provided metadata.
+* An optional "title" which will shown by Auspice when referring to
+  this trait -- for instance you may have a trait called "ab1" which
+  you want to show as "Age bracket 1" in the drop-down menus, legend,
+  and filter.
+* An optional, but highly recommended "type" which can be either
+  'ordinal', 'boolean', 'continuous', or 'categorical'. If you don't
+  provide a type, augur will try to guess it (see how it guesses
+  [here](#id1)).
```
Comment (victorlin) [inline docs/releases/migrating-v5-v6.rst:493]: Use RST link syntax and link to a new anchor for the "Traits" section.

File: docs/faq/what-is-a-build.md [hunk 5]
```diff
@@ -28,5 +28,5 @@ If you look at the [other tutorials](https://docs.nextstrain.org/en/latest/tutor
 While it is possible to run a build by running each of the individual steps, we typically group these together into a make-type file.
 [Snakemake](https://snakemake.readthedocs.io/en/stable/index.html) is "a tool to create reproducible and scalable data analyses... via a human-readable, Python-based language."
 
-> Snakemake is installed as part of the [conda environment](https://docs.nextstrain.org/en/latest/guides/install/local-installation.html#install-augur-auspice-with-conda) or the [docker container](https://docs.nextstrain.org/en/latest/guides/install/cli-install.html).
+> Snakemake is installed as part of the [conda environment](https://docs.nextstrain.org/en/latest/guides/install/local-installation.html) or the [docker container](https://docs.nextstrain.org/en/latest/guides/install/cli-install.html).
```
Comment (victorlin) [inline docs/faq/what-is-a-build.md:31]: _(non-blocking)_

The link doesn't make sense for the link text "conda environment" but I think the only way to address this is to update this entire sentence with newer terminology such as "runtime". That is out of scope of this PR. This change seems appropriate for the sake of satisfying `linkcheck`.

**Review by victorlin — CHANGES_REQUESTED** [review]
I went through all the changes this time so these should be the final tweaks to be made.

File: docs/faq/what-is-a-build.md (hunk 5 above)
Comment (genehack) [reply docs/faq/what-is-a-build.md:31]: Created #1582 to track

**Comment by genehack** [comment]
> I went through all the changes this time so these should be the final tweaks to be made.

All addressed; ready for final check.

**Review by victorlin — APPROVED** [review]
